src/logging/Types.cc

```diff
@@ -0,0 +1,50 @@
+// See the file "COPYING" in the main distribution directory for copyright.
+
+#include "zeek/logging/Types.h"
+
+#include "zeek/Desc.h"
+#include "zeek/Type.h"
+#include "zeek/Val.h"
+
+namespace zeek::logging::detail {
+
+LogWriteHeader::LogWriteHeader() = default;
+
+LogWriteHeader::LogWriteHeader(EnumValPtr arg_stream_id, EnumValPtr arg_writer_id, std::string arg_filter_name,
+                               std::string arg_path)
+    : stream_id(std::move(arg_stream_id)),
+      writer_id(std::move(arg_writer_id)),
+      filter_name(std::move(arg_filter_name)),
+      path(std::move(arg_path)) {
+    stream_name = obj_desc_short(stream_id.get());
+    writer_name = obj_desc_short(writer_id.get());
+}
+
+LogWriteHeader& LogWriteHeader::operator=(const LogWriteHeader& other) = default;
```

Comment:
I want to say I did this because technically the header needs a `Val.h` include to instantiate the defaulted ones, otherwise you get these kind of errors:
```
/home/awelzel/corelight-oss/zeek/src/include/zeek/IntrusivePtr.h:117:17: error: no matching function for call to 'Unref'
  117 |                 Unref(ptr_);
      |                 ^~~~~
/home/awelzel/corelight-oss/zeek/src/include/zeek/logging/Types.h:40:5: note: in instantiation of member function 'zeek::IntrusivePtr<zeek::EnumVal>::~IntrusivePtr' requested here
   40 |     LogWriteHeader() = default;
      |     ^
/home/awelzel/corelight-oss/zeek/src/include/zeek/Obj.h:172:13: note: candidate function not viable: cannot convert argument of incomplete type 'pointer' (aka 'zeek::EnumVal *') to 'Obj *' for 1st argument
  172 | inline void Unref(Obj* o) {
      |             ^     ~~~~~~
1 error generated.
ninja: build stopped: subcommand failed.
```

By moving `= default` into Types.cc, we can do the `#include "zeek/Val.h" just there.

Am I being to avoiding of that "zeek/Val.h" include?

Reply:
I've removed the move constructor, but otherwise left a comment why it's looking like that. If you think that's not sensible, lets chat :-)

Reply:
Gah, I've run into that before. I think what you did is sensible. Resolving.